pwndbg/arguments.py

```diff
@@ -221,9 +222,18 @@ def format_args(instruction):
 
         # Enhance args display
         if arg.name == "fd" and isinstance(value, int):
-            path = pwndbg.gdblib.file.readlink("/proc/%d/fd/%d" % (pwndbg.gdblib.proc.pid, value))
-            if path:
-                pretty += " (%s)" % path
+            if pwndbg.gdblib.proc.pid is not None:
+                path = pwndbg.gdblib.file.readlink(
+                    "/proc/%d/fd/%d" % (pwndbg.gdblib.proc.pid, value)
+                )
+                if path:
+                    pretty += " (%s)" % path
+            else:
+                print(
+                    message.hint(
```

Comment:
Hmm how does this hint message work? When/where does it appear? I mean, we are in a format_args call, maybe we shouldn't log anything at all and just not enhance the path display?

Reply:
Or: we could display it once

Reply:
> Hmm how does this hint message work? When/where does it appear? I mean, we are in a format_args call, maybe we shouldn't log anything at all and just not enhance the path display?

The hint message will display above the context output. Although displaying it below the context would be best, that may be somewhat difficult to implement.
![image](https://user-images.githubusercontent.com/79578430/227685752-879dd326-d151-4bf3-a192-1e2a4b2a0879.png)

Reply:
> Although displaying it below the context would be best, that may be somewhat difficult to implement.

We could probably set some global variable and use it later on, but yeah, meh.

If I think more about it, now, we are displaying the warning but the user has no idea why we do it, do they?

We should either describe it better, saying that "Cannot show the file from fd=5 because we could not find the PID ...", or, remove the warning completely and just not enhance the display.

I think the second one is probably better?

Reply:
> > Although displaying it below the context would be best, that may be somewhat difficult to implement.
> 
> We could probably set some global variable and use it later on, but yeah, meh.
> 
> If I think more about it, now, we are displaying the warning but the user has no idea why we do it, do they?
> 
> We should either describe it better, saying that "Cannot show the file from fd=5 because we could not find the PID ...", or, remove the warning completely and just not enhance the display.
> 
> I think the second one is probably better?

right, I think so.

Reply:
> Ping, can you remove this warning message then and maybe turn it into a comment?

sure, I going to add the comment.